openlr/openlr_simple_decoder.cpp

```diff
@@ -0,0 +1,224 @@
+#include "openlr/openlr_simple_decoder.hpp"
+
+#include "openlr/openlr_model.hpp"
+#include "openlr/openlr_simple_parser.hpp"
+#include "openlr/road_info_getter.hpp"
+#include "openlr/router.hpp"
+#include "openlr/way_point.hpp"
+
+#include "routing/car_model.hpp"
+#include "routing/features_road_graph.hpp"
+#include "routing/road_graph.hpp"
+
+#include "indexer/classificator.hpp"
+#include "indexer/index.hpp"
+
+#include "geometry/polyline2d.hpp"
+
+#include "base/logging.hpp"
+#include "base/math.hpp"
+
+#include "std/algorithm.hpp"
+#include "std/fstream.hpp"
+#include "std/thread.hpp"
+
+using namespace routing;
+
+namespace openlr
+{
+namespace
+{
+size_t constexpr kCacheLineSize = 64;
+
+size_t constexpr GCD(size_t a, size_t b) { return b == 0 ? a : GCD(b, a % b); }
```

Comment:
Make them non-constexpr?
Or, better yet, move the LCM implementation to base/math.hpp.